packages/manager/src/features/CloudView/Overview/DashboardFilterBuilder.tsx

```diff
@@ -0,0 +1,295 @@
+import { Dashboard, TimeDuration } from '@linode/api-v4';
+import KeyboardArrowDownIcon from '@mui/icons-material/KeyboardArrowDown';
+import KeyboardArrowRightIcon from '@mui/icons-material/KeyboardArrowRight';
+import { Grid, Typography } from '@mui/material';
+import * as React from 'react';
+
+import { Button } from 'src/components/Button/Button';
+import { CircleProgress } from 'src/components/CircleProgress';
+import { CloudPulseServiceTypeFilters } from 'src/featureFlags';
+
+import RenderComponent from '../shared/CloudPulseComponentRenderer';
+import { RESOURCES } from '../Utils/CloudPulseConstants';
+import { FILTER_CONFIG } from '../Utils/FilterConfig';
+import { Box } from 'src/components/Box';
+
+export interface DashboardWithFilterProps {
+  dashboard: Dashboard;
+  emitFilterChange: (filterKey: string, value: any) => void;
+  serviceAnalyticsIntegration: boolean;
+}
+
+export const CloudPulseDashboardFilterBuilder = React.memo(
+  (props: DashboardWithFilterProps) => {
+    const [serviceType, setServiceType] = React.useState<string>();
+
+    const [dependentFilters, setDependentFilters] = React.useState<{
+      [key: string]: any;
+    }>({});
+
+    const [showFilter, setShowFilter] = React.useState<boolean>(true);
+
+    const dependentFilterReference: { [key: string]: any } = React.useRef({});
+
+    const handleResourceChange = (resourceId: any) => {
+      props.emitFilterChange('resource_id', resourceId);
+      checkForAndUpdateState('resource_id', resourceId);
+    };
+
+    const handleTimeRangeChange = (
+      start: number,
+      end: number,
+      timeDuration?: TimeDuration,
+      timeRangeLabel?: string
+    ) => {
+      props.emitFilterChange('relative_time_duration', timeDuration);
+      checkForAndUpdateState('relative_time_duration', timeDuration);
+    };
+
+    const handleRegionChange = (region: string | undefined) => {
+      props.emitFilterChange('region', region);
+      checkForAndUpdateState('region', region);
+    };
+
+    const checkForAndUpdateState = (filterkey: string, value: any) => {
+      const serviceTypeConfig = FILTER_CONFIG.get(
+        props.dashboard.service_type!
+      );
+
+      for (
+        let i = 0;
+        i < (serviceTypeConfig ? serviceTypeConfig?.filters.length : 0);
+        i++
+      ) {
+        const filter = serviceTypeConfig?.filters[i];
+        if (
+          filter &&
```

Comment:
```suggestion
          Boolean(filter?.configuration.dependency.length) && filter.configuration.dependency.includes(filterkey)
```

Replace multiple && with single statement